app/models/custom_organisation_name.rb

```diff
@@ -1,3 +1,16 @@
 class CustomOrganisationName < ApplicationRecord
-  validates :name, presence: true, uniqueness: { case_sensitive: false }
+  validate :validate_in_register?
+  validates :name, presence: true
+
+  def validate_in_register?
+    unless Organisation.fetch_organisations_from_register.exclude?(name.strip)
```

Comment:
unless -> exclude could probably be converted to if -> include and be more readable?

Reply:
Im down with this